### PR TITLE
ksmbd: add

### PR DESCRIPTION
**Pull request by neheb**
ksmbd is an upstream linux alternative to Samba which is lighterweight
and more performant, especially on underpowered devices.

Moving here as it is now an upstream kernel module. Also easier to
update as version updates can be coordinated better.

The next LTS kernel (5.15) has this included. A depend on kernel < 5.15
will need to be added later.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

ping @Andy2244 @hauke

### CODE REVIEW

File: package/kernel/ksmbd/Makefile [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+PKG_NAME:=ksmbd
+PKG_VERSION:=3.4.3
+PKG_RELEASE:=$(AUTORELEASE)
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/cifsd-team/cifsd/tar.gz/$(PKG_VERSION)?
+PKG_HASH:=a910c55d9e6924775e00504eddd00b49788603af29d0772cb9fb6722c189f628
+
+PKG_MAINTAINER:=Andy Walsh <andy.walsh44+github@gmail.com>
```
Comment (chunkeey) [inline package/kernel/ksmbd/Makefile:11]: Hmm, should we burden upstream projects with possible openwrt-related requests or not?
(this has just one patch. so possibly?)

**Comment by Neustradamus** [comment]
@openwrt, @neheb, @mans0n, @chunkeey: Any news on it?

ksmbd is linked to the kernel:
- https://www.kernel.org/doc/html/latest/filesystems/cifs/ksmbd.html
- https://samba.plus/blog/detail/ksmbd-a-new-in-kernel-smb-server

Note: there is ksmbd-tools too.

Thanks in advance.

**Comment by chunkeey** [comment]
I keep looking at that PKG_MAINTAINER. If the PR came from Andy Walsh himself (or if he acknowledges it) this is fine. If not, it should be dropped?!

(nitpick subject: "ksmbd: add out-of-tree module" or, "kernel: add standalone ksmbd" , so it fits in with other patches that added kmods.)

**Comment by Andy2244** [comment]
sorry holidays, i'm a bit out of the loop on ksmbd and trust @neheb . 

So the move looks fine to me.

**Comment by chunkeey** [comment]
Ok, that solves that. Thank you!

File: package/kernel/ksmbd/Makefile (hunk 1 above)
Comment (aparcar) [reply package/kernel/ksmbd/Makefile:11]: I'd remove it. No other core package that I'm aware of has non-team maintainers.

**Comment by neheb** [comment]
Removed PKG_MAINTAINER and reworded commit message slightly.

**Comment by Neustradamus** [comment]
@neheb: Nice :)
Hope a quickly merging!

Question for the linked part, how ksmbd-tools will be integrated?

**Comment by neheb** [comment]
It'll stay in the packages feed, just like all the other userspace tools.

**Comment by aparcar** [comment]
> The next LTS kernel (5.15) has this included. A depend on kernel < 5.15 will need to be added later.

Can you please explain this?

**Comment by neheb** [comment]
adding DEPENDS:=@LINUX_5_10 when kernel 5.4 gets removed.

File: package/kernel/ksmbd/Makefile [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+PKG_HASH:=a910c55d9e6924775e00504eddd00b49788603af29d0772cb9fb6722c189f628
+
+PKG_LICENSE:=GPL-2.0-or-later
+PKG_LICENSE_FILES:=COPYING
+
+include $(INCLUDE_DIR)/kernel.mk
+include $(INCLUDE_DIR)/package.mk
+
+TAR_OPTIONS+= --strip-components 1
+TAR_CMD=$(HOST_TAR) -C $(1) $(TAR_OPTIONS)
```
Comment (hauke) [inline package/kernel/ksmbd/Makefile:18]: Why are these tar options needed?

File: package/kernel/ksmbd/Makefile [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		+kmod-crypto-sha512 \
+		+kmod-crypto-aead \
+		+kmod-crypto-ccm \
+		+kmod-crypto-gcm \
+		+kmod-fs-nfsd \
+		+kmod-nf-nathelper-extra
+endef
+
+# The last two DEPENDS are hacks in order to get CONFIG_ASN1 and CONFIG_OID_REGISTRY
+# which it seems can't be selected independently. Some bug in either base or upstream.
```
Comment (hauke) [inline package/kernel/ksmbd/Makefile:44]: The ASN1 configuration option dos not define an help text, then you can not directly select it:
https://github.com/torvalds/linux/blob/master/init/Kconfig#L2362

You could add a help text, then it is possible to select it. We do it for some configuration options here: https://github.com/openwrt/openwrt/blob/master/target/linux/generic/hack-5.10/251-kconfig.patch

**Comment by Neustradamus** [comment]
@neheb: Have you seen @hauke comments?

**Comment by neheb** [comment]
I seem to have messed the branch up. Will resubmit.

**Pull request closed without merging**